src/lib/i18n.jsx

```diff
@@ -41,11 +41,12 @@ export function getDefaultLanguage(available, override) {
 }
 
 export function LanguageProvider({ children }) {
-	const location = useLocation();
+	const { query } = useLocation();
 
 	const [lang, setLang] = useStoredValue(
 		'lang',
-		getDefaultLanguage(config.languages, location.query.lang) || 'en'
+		getDefaultLanguage(config.languages, query.lang) || 'en',
+		!!query.lang
```

Comment:
Problem with the existing implementation (IMO) is that the query param would not override a set value in localstorage, which feels wrong.